src/dynamics/joints/mod.rs

```diff
@@ -78,6 +78,11 @@ pub trait JointDef {
 
     #[doc(hidden)]
     unsafe fn create<U: UserDataTypes>(&self, world: &mut World<U>) -> *mut ffi::Joint;
+    
+    #[doc(hidden)]
+    unsafe fn try_create<U: UserDataTypes>(&self, world: &mut World<U>) -> Option<*mut ffi::Joint> {
+        Some(self.create(world))
```

Comment:
I would define `create` in terms of `try_create` to avoid code duplication?

Reply:
I edited all implementations of `create` to be in terms of `try_create`.

I made the trait default for `try_create` be implemented in terms of `create` because the trait is public, so any users implementing `JointDef` would have their implementation break without this default.

If this sort of breaking change is not a concern for this project I can re-define the trait defaults so `create` is in terms of `try_create`, and then the implementers would only need the `try_create` function.

Reply:
That makes sense, I guess we can make this breaking change at another point in time when there is a bigger version bump.